src/main/java/competition/subsystems/drive/commands/DriveToPositionCommand.java

```diff
@@ -3,33 +3,40 @@
 import javax.inject.Inject;
 
 import xbot.common.command.BaseCommand;
+import xbot.common.math.PIDManager;
+import xbot.common.math.PIDManager.PIDManagerFactory;
 import xbot.common.properties.DoubleProperty;
 import xbot.common.properties.PropertyFactory;
 import competition.subsystems.drive.DriveSubsystem;
 import competition.subsystems.pose.PoseSubsystem;
-import edu.wpi.first.units.measure.Velocity;
 
 public class DriveToPositionCommand extends BaseCommand {
 
-    DriveSubsystem drive;
-    PoseSubsystem pose;
+    // Subsystems
+    final DriveSubsystem drive;
+    final PoseSubsystem pose;
+    final PIDManager pid;
 
+    // Physics computation numbers
     double targetPosition;
     double previousPosition;
 
     double error;
     double velocity;
 
-    DoubleProperty proportionalConstant;
-    DoubleProperty derivativeConstant;
-
     @Inject
-    public DriveToPositionCommand(DriveSubsystem driveSubsystem, PoseSubsystem pose, PropertyFactory propertyFactory) {
+    public DriveToPositionCommand(DriveSubsystem driveSubsystem, PoseSubsystem pose, PIDManagerFactory pidManagerFactory) {
         this.drive = driveSubsystem;
         this.pose = pose;
-        propertyFactory.setPrefix(this);
-        proportionalConstant = propertyFactory.createPersistentProperty("Proportional constant (error)", 1);
-        derivativeConstant = propertyFactory.createPersistentProperty("Derivative constant (velocity)", 1);
+        this.pid = pidManagerFactory.create("DriveToPoint");
+
+        pid.setEnableErrorThreshold(true); // Turn on distance checking
```

Comment:
all these values can be set as args in the `create()` call above if you want something more concise (but this is functionally fine too)